TACC/Core-Styles
# feat: des-2697 s-app-card & bootstrap 3 has-row--col-stretch-content-y

## Overview

Style "app cards" for (initially for DesignSafe client).

> [!IMPORTANT]
> This is a structure only UI pattern. [DesignSafe client will import and skin it.](https://github.com/DesignSafe-CI/portal/pull/1168)

> [!NOTE]
> **To Do**
> - [x] make demo responsive, not flexible

## Related

- [DES-2697](https://tacc-main.atlassian.net/browse/DES-2697)
- https://github.com/DesignSafe-CI/portal/pull/1168

## Changes

- **added** `.s-app-card` a.k.a `.app-card`
- **fixed** `.has-row--col-stretch-content-y` in Bootstrap 3

## Testing

1. Open http://localhost:3000/components/preview/s-app-card.
2. See how un-skinned it looks.
3. But notice it has a structure / layout.
4. Compare to [skinned version](https://github.com/DesignSafe-CI/portal/pull/1168).

## UI

### Static

| Raw Screenshot | Highlighted Grid & Flex |
| - | - |
| <img width="960" alt="structure - unhighlighted" src="https://github.com/TACC/Core-Styles/assets/62723358/e22cc1a1-3acb-4af3-be68-4bc66015f52a"> | <img width="960" alt="structure - highlighted" src="https://github.com/TACC/Core-Styles/assets/62723358/0c100819-f76d-4fb7-8a5b-6de71d7c9a64"> |

### Responsiveness

https://github.com/TACC/Core-Styles/assets/62723358/58042f40-2f8c-4b5f-8937-508c50993712

<details><summary><h3>Client Use Case</h3></summary>

https://github.com/DesignSafe-CI/portal/assets/62723358/fd525337-7c83-4f5e-98c5-9ee5bdb73db4

</details>
